SCons/Tool/msvc.py

```diff
@@ -236,7 +256,8 @@ def generate(env):
         shared_obj.add_emitter(suffix, shared_object_emitter)
 
     env['CCPDBFLAGS'] = SCons.Util.CLVar(['${(PDB and "/Z7") or ""}'])
-    env['CCPCHFLAGS'] = SCons.Util.CLVar(['${(PCH and "/Yu%s \\\"/Fp%s\\\""%(PCHSTOP or "",File(PCH))) or ""}'])
+    env['CCPCHFLAGS_GEN'] = CCPCHFLAGS_gen
```

Comment:
why do this? The current expression matches your function?

Reply:
I had a comment about this: https://github.com/SCons/scons/pull/3984#discussion_r675718333

it will maintain compatibility fi someone is adding a string to the variable was my assumption